starknet_py/tests/e2e/docs/code_examples/test_gateway_client.py

```diff
@@ -0,0 +1,207 @@
+# pylint: disable=unused-variable, redefined-builtin
+import pytest
+from starkware.starknet.public.abi import (
+    get_selector_from_name,
+    get_storage_var_address,
+)
+
+from starknet_py.net.client_models import Call
+from starknet_py.net.gateway_client import GatewayClient
+from starknet_py.net.networks import TESTNET
+from starknet_py.tests.e2e.fixtures.constants import MAX_FEE
+
+
+def test_init():
+    # docs-start: init
+    gateway_client = GatewayClient(net=TESTNET)
+    # or (for custom urls)
+    gateway_client = GatewayClient(
+        net={"feeder_gateway_url": "some_feeder_url", "gateway_url": "some_gateway_url"}
+    )
+    # docs-end: init
+
+
+@pytest.mark.asyncio
+async def test_get_block(gateway_client):
+    # docs-start: get_block
+    block = await gateway_client.get_block(block_hash="latest")
+    # or
+    block = await gateway_client.get_block(block_number=0)
+    # docs-end: get_block
+
+
+@pytest.mark.asyncio
+async def test_get_block_traces(gateway_client):
+    # docs-start: get_block_traces
+    block_traces = await gateway_client.get_block_traces(block_hash="latest")
+    # or
+    block_traces = await gateway_client.get_block_traces(block_number=0)
+    # docs-end: get_block_traces_end
+
+
+@pytest.mark.asyncio
+async def test_get_state_update(gateway_client):
+    # docs-start: get_state_update_start
+    state_update = await gateway_client.get_state_update(block_hash="latest")
+    # or
+    state_update = await gateway_client.get_state_update(block_number=0)
+    # docs-end: get_state_update
+
+
+@pytest.mark.asyncio
+async def test_get_storage_at(gateway_client, map_contract):
+    address = map_contract.address
+    # docs-start: get_storage_at
+    state_update = await gateway_client.get_storage_at(
+        contract_address=address,
+        key=get_storage_var_address("storage_var name"),
+        block_hash="latest",
+    )
+    # docs-end: get_storage_at
+
+
+@pytest.mark.asyncio
+async def test_get_transaction(gateway_client, declare_transaction_hash):
+    transaction_hash = declare_transaction_hash
+    # docs-start: get_transaction
+    transaction = await gateway_client.get_transaction(tx_hash=transaction_hash)
+    # docs-end: get_transaction
+
+
+@pytest.mark.asyncio
+async def test_get_transaction_receipt(gateway_client, declare_transaction_hash):
+    transaction_hash = declare_transaction_hash
+    # docs-start: get_transaction_receipt
+    transaction = await gateway_client.get_transaction_receipt(tx_hash=transaction_hash)
+    # docs-end: get_transaction_receipt
+
+
+@pytest.mark.asyncio
+async def test_estimate_fee(gateway_account, deploy_account_transaction):
+    gateway_client = gateway_account.client
+    transaction = deploy_account_transaction
+    # docs-start: estimate_fee
+    # The transaction can be of type  Invoke | Declare | DeployAccount
```

Comment:
Is there a need for this comment? It creates a possibility of getting out of date if things change as it's in an unusual place